[Do not merge] For tests -- #192 test failure

ekes opened the pull request.

ekes committed:
NOT TO MERGE. Testing screenshots as artifacts.

ekes committed:
not in a subdir.

ekes committed:
Maybe in the correct place.

ekes committed:
Maybe as a seperate job, otherwise, maybe need to fail differently.

stephen-cox committed:
Save files if PHPUnit tests fail

stephen-cox commented:
@ekes I adjusted the GitHub config and the screenshots are now available to download at the bottom of this page: https://github.com/localgovdrupal/localgov_services/actions/runs/3940524749

ekes committed:
What's the difference between keyPress and keyDown?

ekes committed:
See if we can remove the previous results first.

The keyPress keyDown keyUp don't make any visible difference in the
field in Lando; but keyDown (not others) does seem to make a difference
to the timing. The waitOnAutocomplete sadly doesn't :-/
Still this doesn't break running the tests in Lando. Does it help
github?

ekes committed:
Fixing tests for PHP 7 might be another support branch another question.

ekes commented:
Follow-up #195

ekes commented:
For me of the future searching for how we saved the screenshots from phpunit here:

After:
```
      - name: Run PHPUnit tests
        run: |
          mkdir -p ./html/web/sites/simpletest && chmod 777 ./html/web/sites/simpletest
          sed -i "s#http://localgov.lndo.site#http://drupal#" ./html/phpunit.xml.dist
          docker exec -t drupal bash -c 'chown docker:docker -R /var/www/html'
          docker exec -u docker -t drupal bash -c "cd /var/www/html && ./bin/paratest --processes=4 ${{ env.LOCALGOV_DRUPAL_PROJECT_PATH }}"
```

Came:
```
      - name: Save files, screenshots, placed in the files dir
        if: ${{ failure() }}
        uses: actions/upload-artifact@v3
        with:
          name: test-files-dir
          path: html/web/sites/default/files/
```
